app/src/androidTest/java/org/mozilla/focus/activity/PrivateBrowsingTest.java

```diff
@@ -0,0 +1,134 @@
+    @Rule
+    public final ActivityTestRule<MainActivity> activityRule = new ActivityTestRule<>(MainActivity.class, true, false);
+
+    @Before
+    public void setUp() {
+        // Set the share preferences and start the activity
+        AndroidTestUtils.beforeTest();
+        tabsDatabase = Inject.getTabsDatabase(null);
+        GeoPermissionCache.putAllowed(null, Boolean.FALSE);
+    }
+
+    /**
+     * Test case no: TC0108
+     * Test case name: Open private mode and back when browser tab number is 0
+     * Steps:
+     * 1. Launch app
+     * 2. Tap menu
+     * 3. Tap private mode
+     * 4. Tap private button to be back normal mode
+     * 5. Check tab tray number is 0
+     */
+    @Test
+    public void openPrivateModeAndBack_whenBrowserTabNumberZero() {
+
+        // Start activity
+        activityRule.launchActivity(new Intent());
+
+        // Tap menu
+        AndroidTestUtils.tapBrowserMenuButton();
+
+        // Tap private mode
+        AndroidTestUtils.tapPrivateButtonInMenu();
+
+        // Tap private button to be back normal mode
+        AndroidTestUtils.tapPrivateButtonBackToBrowser();
+
+        // Check tab tray number is 0
+        onView(allOf(withId(R.id.counter_text), isDescendantOfA(withId(R.id.home_screen_menu)))).check(matches(withText("0")));
+    }
+
+    /**
+     * Test case no: TC0109
+     * Test case name: Open private mode and back when browser tab number is over 0
+     * Steps:
+     * 1. Launch app with tab number is 2 (by searching site_1 and then site2)
+     * 2. Tap menu -> private mode
+     * 3. Tap private button to be back normal mode
+     * 4. Check tab number is 2
+     * 5. Tap tab tray number
+     * 6. Check top item (site_2) focused
+     * 7. Tab private mode in tab tray
+     * 8. Tab system back key
+     * 9. Check top item (site_2) focused
+     */
+    @Test
+    public void openPrivateModeAndBack_whenBrowserTabNumbeOverZero() {
+
+        // Launch Rocket with tab number is 2 (by searching site_1 and then site2)
+        tabsDatabase.tabDao().insertTabs(TAB, TAB_2);
+        AndroidTestUtils.setFocusTabId(TAB.getId());
+        activityRule.launchActivity(new Intent());
+
+        // Tap menu -> private mode
+        AndroidTestUtils.tapBrowserMenuButton();
+        AndroidTestUtils.tapPrivateButtonInMenu();
+
+        // Tap private button to be back normal mode
+        AndroidTestUtils.tapPrivateButtonBackToBrowser();
+
+        // Check tab number is 2 and tap it
+        onView(allOf(withId(R.id.counter_text), isDescendantOfA(withId(R.id.browser_screen_menu)))).check(matches(withText("2"))).perform(click());
+
+        // Check top item (site_2) focused
+        onView(withId(R.id.tab_tray)).check(matches(atPosition(0, isSelected())));
+
+        // Tab private mode in tab tray
+        onView(allOf(withId(R.id.btn_private_browsing), isDisplayed())).perform(click());
```

Comment:
I'm confused. You had tapPrivateButtonInMenu in your AndroidTestUtils.ja, and this lin also clicks the same UI.
How do you prevent duplicating those calls everywhere?